src/PIL/WebPImagePlugin.py

```diff
@@ -240,16 +242,23 @@ def _save_all(im, fp, filename):
 
                 # Make sure image mode is supported
                 frame = ims
+                rawmode = ims.mode
                 if ims.mode not in _VALID_WEBP_MODES:
-                    alpha = ims.mode == 'P' and 'A' in ims.im.getpalettemode()
-                    frame = ims.convert('RGBA' if alpha else 'RGBX')
+                    alpha = 'A' in ims.mode or 'a' in ims.mode \
+                            or (ims.mode == 'P' and 'A' in ims.im.getpalettemode())
+                    rawmode = 'RGBA' if alpha else 'RGBX'
```

Comment:
Is it possible to use `RGB` instead? I'm not sure, probably `ims.convert('RGB`)` has more wide range of supported source modes, than `ims.convert('RGBX`)`. There should no be difference in other aspects.

Reply:
That seems to make sense. I made the change. Few lines later, when we exporting to bytes it still uses `RGBX` rawmode.